src/lib/storage/table.cpp

```diff
@@ -191,6 +191,24 @@ std::vector<bool> Table::columns_are_nullable() const {
   return nullable;
 }
 
+bool Table::column_is_unique(const ColumnID column_id) const {
+  Assert(column_id < _column_definitions.size(), "ColumnID out of range.");
+  for (const auto& key_constraint : soft_key_constraints()) {
+    const auto& key_type = key_constraint.key_type();
+    if (key_type != KeyConstraintType::PRIMARY_KEY && key_type != KeyConstraintType::UNIQUE) {
+      continue;
+    }
+
+    for (const auto& constraint_column_id : key_constraint.columns()) {
+      if (constraint_column_id == column_id) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
```

Comment:
Please move to `constraint_utils.cpp`

Reply:
Yes, you are missing knowledge about upcoming things. :)
You are right that it currently would be better placed in the table. However, we plan to move constraints (at some point, NULLing will be a constraint) away from the tables. Hence the placement in the utils.